server/src/test/java/org/elasticsearch/common/xcontent/support/XContentMapValuesTests.java

```diff
@@ -302,6 +301,16 @@ public void testExtractRawValueMixedObjects() throws IOException {
                 assertThat(XContentMapValues.extractRawValues("foo.bar", map), Matchers.containsInAnyOrder("meow", "baz"));
             }
         }
+        {
+            XContentBuilder builder = XContentFactory.jsonBuilder().startObject()
+                .field("foo", "bar")
+                .field("foo.subfoo", "baz")
+                .endObject();
+            try (XContentParser parser = createParser(JsonXContent.jsonXContent, Strings.toString(builder))) {
+                assertThat(XContentMapValues.extractRawValues("foo.subfoo", parser.map()),
+                    containsInAnyOrder("baz"));
```

Comment:
were we always returning also foo's values or only when foo.subfoo did not exist?

Reply:
Always both of them.

Reply:
oh boy, it looks like a bad bug at first glance, on the other hand I am thinking that given that we used to dynamically map objects in the properties section, it is not so easy to hit it, yet good to fix it now that objects are no longer mapped when dynamic is set to runtime which allows for even more free mappings.